packages/tutorials/election-dapp/front-end/src/Candidate/Candidate.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from 'react'
+import { getVotes } from '../contract'
+import './Candidate.css'
+
+export interface ICandidate {
+  id: string
+  name: string
+}
+
+interface IProps {
+  candidate: ICandidate
+  voteAllowed: boolean
+  voteInProgress: boolean
+  onVote: (candidateId: string) => void
+}
+
+export const Candidate: React.FC<IProps> = ({ candidate: { id, name }, voteAllowed, voteInProgress, onVote }): JSX.Element => {
+  const [voteCount, setVoteCount] = useState<number>(0)
+
+  const retrieveCandidateVotes = async (id: string) => {
+    const votes = await getVotes(id)
+    setVoteCount(votes)
+  }
+
+  useEffect(() => {
+    if (!voteInProgress) {
+      retrieveCandidateVotes(id)
+    }
+  }, [id, voteInProgress])
+
+  return (
+    <div className="Candidate-row">
```

Comment:
Can we make this more semantic?
e.g. `section`

Reply:
Yes, done.